src/dbnode/storage/namespace.go

```diff
@@ -1641,57 +1640,99 @@ func (n *dbNamespace) aggregateTiles(
 		SchemaDesc:   nsCtx.Schema,
 	}
 
-	resources, err := newColdFlushReuseableResources(n.opts)
-	if err != nil {
-		return 0, err
-	}
-
-	var blockReaders []fs.DataFileSetReader
 	var sourceBlockStarts []time.Time
 	sourceBlockSize := sourceNs.Options().RetentionOptions().BlockSize()
 	for sourceBlockStart := blockStart; sourceBlockStart.Before(opts.End); sourceBlockStart = sourceBlockStart.Add(sourceBlockSize) {
-		reader, err := fs.NewReader(sourceNs.StorageOptions().BytesPool(), sourceNs.StorageOptions().CommitLogOptions().FilesystemOptions())
-		if err != nil {
-			return 0, err
-		}
 		sourceBlockStarts = append(sourceBlockStarts, sourceBlockStart)
-		blockReaders = append(blockReaders, reader)
 	}
 
+	targetShards := n.OwnedShards()
+	targetShardIndexQueue := make(chan int, len(targetShards))
```

Comment:
Would it be better to use a channel of pointers to shards, instead of shard slice indices?